prov/rxm/src/rxm_init.c

```diff
@@ -184,6 +184,9 @@ int rxm_info_to_rxm(uint32_t version, const struct fi_info *core_info,
 			return -FI_ENOMEM;
 	}
 
+	ofi_alter_mr_mode(&info->domain_attr->mr_mode,
+			  core_info->domain_attr->mr_mode, info->caps, version);
+
```

Comment:
info->domain_attr->mr_mode is being modified at line 176, then again here.  There should be a consistent place for updating mr_mode.  Is the mr_mode from the core_info coming back incorrectly set?

Assuming that isn't the case, why is info-> ... ->mr_mode incorrectly set at this point?  Is there a problem earlier in the code flow where this fix is needed, rather than trying to clean it up at the end?

Reply:
Can you also test against the latest master to verify that the problem still exists?  A change went in this week that modified the util provider getinfo path.  Walking through the code, ofix_getinfo() calls ofi_alter_info(), which should adjust the mr_mode.  Is there a problem in that path?

Reply:
I am still seeing the issue. But, it appears the the verbs MR mode bits are set incorrectly. That seems to be the correct location for the fix.

    fi_domain_attr:
        domain: 0x0
        name: mlx4_0
        threading: FI_THREAD_SAFE
        control_progress: FI_PROGRESS_AUTO
        data_progress: FI_PROGRESS_AUTO
        resource_mgmt: FI_RM_ENABLED
        av_type: FI_AV_UNSPEC
        mr_mode: [ FI_MR_BASIC, FI_MR_LOCAL, FI_MR_VIRT_ADDR, FI_MR_ALLOCATED, FI_MR_PROV_KEY ]
        mr_key_size: 4
        cq_data_size: 0
        cq_cnt: 65408
        ep_cnt: 393144
        tx_ctx_cnt: 1024
        rx_ctx_cnt: 1024
        max_ep_tx_ctx: 1024
        max_ep_rx_ctx: 1024
        max_ep_stx_ctx: 0
        max_ep_srx_ctx: 65472
        cntr_cnt: 0
        mr_iov_limit: 1
    caps: [ FI_LOCAL_COMM, FI_REMOTE_COMM ]
    mode: [  ]
        auth_key_size: 0
        max_err_data: 255
        mr_cnt: 524032
    fi_fabric_attr:
        name: IB-0xfe80000000000000
        prov_name: verbs
        prov_version: 110.0
        api_version: 1.10